atomix/cluster/src/main/java/io/atomix/raft/storage/log/DelayedFlusher.java

```diff
@@ -39,37 +38,33 @@ public DelayedFlusher(final Scheduler scheduler, final Duration delayTime) {
 
   @Override
   public void flush(final Journal journal) {
-    scheduleFlush(journal);
+    synchronized (scheduledMonitor) {
+      if (scheduledFlush == null) {
+        scheduledFlush = scheduler.schedule(delayTime, () -> asyncFlush(journal));
+      }
+    }
   }
 
   @Override
   public void close() {
-    if (closed) {
```

Comment:
I think the close method is idempotent already anyway, and it's always called from the same thread. Checking if we're closed in the `asyncFlush` method was just a best effort anyway, so we can omit it.